web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EksClustersList.tsx

```diff
@@ -121,26 +123,51 @@ function getStatus(item: CheckedEksCluster) {
   }
 }
 
-function disabledStates(item: CheckedEksCluster, autoDiscovery: boolean) {
-  const disabled =
-    getStatus(item) !== ItemStatus.Success ||
-    item.kubeServerExists ||
-    autoDiscovery;
+function disabledStates(
+  item: CheckedEksCluster,
+  autoDiscovery: boolean
+): { disabled: boolean; disabledText: string } {
+  if (autoDiscovery) {
+    return {
+      disabled: true,
+      disabledText: 'All eligible EKS clusters will be enrolled automatically',
+    };
+  }
 
-  let disabledText = `This EKS cluster is already enrolled and is a part of this cluster`;
-  switch (item.status) {
-    case 'failed':
-    case 'pending':
-    case 'creating':
-    case 'updating':
-      disabledText = 'Not available, try refreshing the list';
-      break;
-    case 'deleting':
-      disabledText = 'Not available';
+  if (item.kubeServerExists) {
+    return {
+      disabled: true,
+      disabledText:
+        'This EKS cluster is already enrolled and is a part of this cluster',
+    };
   }
-  if (autoDiscovery) {
-    disabledText = 'All eligible EKS clusters will be enrolled automatically';
+
+  if (cfg.isCloud && !item.endpointPublicAddress) {
+    return {
+      disabled: true,
+      disabledText:
+        'Please enable endpoint public access in your EKS cluster and try again.',
+    };
+  }
+
+  if (item.authenticationMode === 'CONFIG_MAP') {
+    return {
+      disabled: true,
+      disabledText:
+        'Authentication mode is CONFIG_MAP, only API and API_AND_CONFIG_MAP are supported.',
+    };
+  }
```

Comment:
should we do `!= API && != API_AND_CONFIG_MAP` isntead?

Reply:
Given that we only set 3 possible values, I don't think it would be any different. But I'll change it to be more explicit about the valid values 👍